server/controllers/Profile.js

```diff
@@ -0,0 +1,76 @@
+const mongoose = require("mongoose");
+const Profile = require("../Models/Profile");
+const asyncHandler = require("express-async-handler");
+
+// @route GET /profile
+// @desc the profile of the relevant user
+// @access Private
+exports.getProfile = asyncHandler(async (req, res, next) => {
+    const userId = req.user.id;
+
+    const profile = await Profile.findOne({ userId });
+
+    res.send({ profile });
+});
+
+// @route POST /profile
+// @desc Create a new profile
+// @access Private
+exports.createProfile = asyncHandler(async (req, res, next) => {
+    const userId = req.user.id;
+
+    const { firstName, lastName, email, phoneNum } = req.body;

```

Comment:
You can first validate all these fields and return a bad request status if any of them is required but missing in the request body.